src/sql/platform/accounts/common/accountStore.ts

```diff
@@ -43,10 +48,37 @@ export default class AccountStore implements IAccountStore {
 
 	public getAllAccounts(): Thenable<azdata.Account[]> {
 		return this.doOperation(() => {
-			return this.readFromMemento();
+			return this.cleanupDeprecatedAccounts().then(() => {
+				return this.readFromMemento();
+			});
 		});
 	}
 
+	public cleanupDeprecatedAccounts(): Thenable<void> {
+		return this.readFromMemento()
+			.then(accounts => {
+				// No need to waste cycles
+				if (!accounts || accounts.length === 0) {
+					return Promise.resolve();
+				}
+				// Remove old accounts that are now deprecated
+				try {
+					accounts = accounts.filter(account => {
+						const providerKey = account?.key?.providerId;
+						// Account has no provider, remove it.
+						if (providerKey === undefined) {
+							return false;
+						}
+						// Returns true if the account isn't from a deprecated provider
+						return !this.deprecatedProviders.includes(account.key.providerId);
```

Comment:
nit:`account.key.providerId` could be `providerKey`

Reply:
thanks!